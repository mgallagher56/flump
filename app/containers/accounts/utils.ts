import type { Account } from './types';

export enum AccountTypeEnum {
  CURRENT = 'Current',
  SAVING = 'Saving',
  MORTGAGE = 'Mortgage',
  LOAN = 'Loan',
  OWED = 'Owed'
}

export const isAccountTypeValid = (accountType: AccountTypeEnum, accounts: Account[]) =>
  !!accounts?.map(({ type }) => type).some((type) => type === accountType);
