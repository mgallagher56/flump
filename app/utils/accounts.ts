export const monthYearSort = (
  a: {
    month?: number;
    year?: number;
  },
  b: {
    month?: number;
    year?: number;
  }
) => {
  if (a.year === b.year) {
    return a.month - b.month;
  }
  return a.year - b.year;
};
