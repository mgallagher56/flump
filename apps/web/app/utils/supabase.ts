import { createBrowserClient, createServerClient, parseCookieHeader, serializeCookieHeader } from '@supabase/ssr';

import type { Database } from 'db_types';

declare global {
  interface Window {
    env: {
      SUPABASE_URL: string;
      SUPABASE_ANON_KEY: string;
    };
  }
}

const isServer = typeof window === 'undefined';

export const getSupaBaseUrl = (isServer: boolean): string =>
  isServer ? process.env.SUPABASE_URL : window?.env?.SUPABASE_URL;
export const getSupabaseAnonKey = (isServer: boolean): string =>
  isServer ? process.env.SUPABASE_ANON_KEY : window?.env?.SUPABASE_ANON_KEY;

const supabase = createBrowserClient<Database>(getSupaBaseUrl(isServer), getSupabaseAnonKey(isServer));

export const createSupaBaseServerClient = (request: Request) => {
  const headers = new Headers();

  return createServerClient<Database>(getSupaBaseUrl(isServer), getSupabaseAnonKey(isServer), {
    cookies: {
      getAll() {
        return parseCookieHeader(request.headers.get('Cookie') ?? '');
      },
      setAll(cookiesToSet) {
        cookiesToSet.forEach(({ name, value, options }) => {
          headers.append('Set-Cookie', serializeCookieHeader(name, value, options));
        });
      }
    }
  });
};

export default supabase;
