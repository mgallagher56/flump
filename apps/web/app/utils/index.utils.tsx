import type { ReactElement } from 'react';

import type { TFunction } from 'i18next';
import FLPBox from '~/components/core/structure/FLPBox';
import type { TabData } from '~/components/core/tabs/types';
import FLPHeading from '~/components/core/typography/FLPHeading';
import type { Employee } from '~/routes/_index';

import { FallbackEnums } from './utils';

const { NA } = FallbackEnums;

export const getTabsData = (employees: Partial<Employee>[], t: TFunction): TabData[] => [
  {
    label: t('featureNum', { num: 1 }),
    value: 'feature-1',
    children: (
      <main>
        <FLPBox>
          <FLPHeading as="h1">{t('featureNum', { num: 1 })}</FLPHeading>
          {employees.map(
            (employee, index): ReactElement => (
              <FLPBox key={employee.id ?? index} style={{ display: 'flex', gap: '.5rem' }}>
                <h3>{`${employee.name ?? NA} - ${employee.department ?? NA}`}</h3>
              </FLPBox>
            )
          )}
        </FLPBox>
      </main>
    )
  },
  {
    label: t('featureNum', { num: 2 }),
    value: 'feature 2',
    children: (
      <main>
        <FLPBox>
          <FLPHeading as="h1">{t('featureNum', { num: 2 })}</FLPHeading>
        </FLPBox>
      </main>
    )
  },
  {
    label: t('featureNum', { num: 3 }),
    value: 'feature-3',
    children: (
      <main>
        <FLPBox>
          <FLPHeading as="h1">{t('featureNum', { num: 3 })}</FLPHeading>
        </FLPBox>
      </main>
    )
  }
];
