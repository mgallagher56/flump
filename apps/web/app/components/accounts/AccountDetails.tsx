import { Fragment, useCallback } from 'react';
import type { ChangeEvent, FC } from 'react';

import { Stack } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { useLoaderData, useRevalidator } from 'react-router';
import FLPButton from '~/components/core/buttons/FLPButton';
import FLPInput from '~/components/core/inputs/input/FLPInput';
import FLPHeading from '~/components/core/typography/FLPHeading';
import FLPText from '~/components/core/typography/FLPText';
import type { loader } from '~/routes/app.accounts.$account';
import supabase from '~/utils/supabase';

import { accountDetailDisplayStyles } from './AccountDetailsStyles';

interface AccountDetailsProps {
  onInputChange: (event: ChangeEvent<HTMLInputElement>) => void;
  editedValues: { [key: string]: { [key: string]: string } };
  isEditMode: boolean;
}

const AccountDetails: FC<AccountDetailsProps> = ({ onInputChange, editedValues, isEditMode }) => {
  const { t } = useTranslation();
  const { account, accountDetails } = useLoaderData<typeof loader>();
  const { revalidate } = useRevalidator();

  const onRemoveYear = useCallback(
    async (year: number) => {
      await supabase.from('account_details').delete().eq('account_id', account.id).eq('year', year).select();
      revalidate();
    },
    [account.id, revalidate]
  );

  return (
    <Stack flexDirection="column" gap={5} overflow="auto">
      {accountDetails
        .map(({ year }: { year?: number }) => year)
        .filter(
          (
            value: number,
            index: number,
            self: {
              indexOf: (arg0: number) => number;
            }
          ) => self.indexOf(value) === index
        )
        .map((year: number) => (
          <Fragment key={year}>
            <Stack alignItems="center" flexDirection="row">
              <FLPHeading as="h3" size="lg">
                {year}
              </FLPHeading>
              <FLPButton colorScheme="red" size="xs" variant="outline" onClick={() => onRemoveYear(year)}>
                {t('deleteYear')}
              </FLPButton>
            </Stack>
            <div className={accountDetailDisplayStyles}>
              {accountDetails.map(
                ({ month, year: currentYear, value }: { month?: number; year?: number; value?: number }) => {
                  if (year !== currentYear) return null;
                  const monthName = new Date(0, month - 1).toLocaleString('default', { month: 'long' });
                  return (
                    <div key={`${year}-${month}`}>
                      <FLPText>{monthName}</FLPText>
                      {isEditMode ? (
                        <FLPInput
                          isLabelHidden
                          data-month={month}
                          data-year={year}
                          label={`${year}-${monthName}`}
                          name="amount"
                          placeholder="0.00"
                          type="number"
                          value={editedValues?.[year]?.[month] ?? value}
                          width={100}
                          onChange={onInputChange}
                        />
                      ) : (
                        <FLPText>
                          {Intl.NumberFormat('en-GB', { style: 'currency', currency: 'GBP' })
                            .format(value)
                            .slice(0, -3)}
                        </FLPText>
                      )}
                    </div>
                  );
                }
              )}
            </div>
          </Fragment>
        ))}
    </Stack>
  );
};

export default AccountDetails;
