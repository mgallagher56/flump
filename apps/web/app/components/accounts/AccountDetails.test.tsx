import { fireEvent, render } from '@testing-library/react';
import { vi } from 'vitest';
import { emptyObject } from '~/utils/utils';

import AccountDetails from './AccountDetails';

const mocks = vi.hoisted(() => ({
  mockFrom: vi.fn(() => ({
    delete: vi.fn(() => ({
      eq: vi.fn(() => ({
        eq: vi.fn(() => ({}))
      }))
    }))
  })),
  mockAccountDetails: [
    {
      month: 1,
      year: 2021,
      value: 1000
    },
    {
      month: 2,
      year: 2021,
      value: 1000
    },
    {
      month: 3,
      year: 2021,
      value: 1000
    },
    {
      month: 4,
      year: 2021,
      value: 1000
    },
    {
      month: 5,
      year: 2021,
      value: 1000
    },
    {
      month: 6,
      year: 2021,
      value: 1000
    },
    {
      month: 7,
      year: 2021,
      value: 1000
    },
    {
      month: 8,
      year: 2021,
      value: 1000
    },
    {
      month: 9,
      year: 2021,
      value: 1000
    },
    {
      month: 10,
      year: 2021,
      value: 1000
    },
    {
      month: 11,
      year: 2021,
      value: 1000
    },
    {
      month: 12,
      year: 2021,
      value: 1000
    },
    {
      month: 1,
      year: 2023,
      value: 1000
    },
    {
      month: 2,
      year: 2023,
      value: 1000
    },
    {
      month: 3,
      year: 2023,
      value: 1000
    },
    {
      month: 4,
      year: 2023,
      value: 1000
    },
    {
      month: 5,
      year: 2023,
      value: 1000
    },
    {
      month: 6,
      year: 2023,
      value: 1000
    },
    {
      month: 7,
      year: 2023,
      value: 1000
    },
    {
      month: 8,
      year: 2023,
      value: 1000
    },
    {
      month: 9,
      year: 2023,
      value: 1000
    },
    {
      month: 10,
      year: 2023,
      value: 1000
    },
    {
      month: 11,
      year: 2023,
      value: 1000
    },
    {
      month: 12,
      year: 2023,
      value: 1000
    }
  ]
}));

vi.mock('react-router', async () => {
  const actual: Record<string, unknown> = await vi.importActual('react-router');
  return {
    ...actual,
    useLoaderData: () => ({ accountDetails: mocks.mockAccountDetails, account: { id: '123456' } }),
    useRevalidator: () => ({ revalidate: vi.fn() })
  };
});

vi.mock('app/utils/supabase', () => ({
  default: {
    from: mocks.mockFrom
  }
}));

describe('<AccountDetails />', () => {
  test('should render', () => {
    const { container } = render(
      <AccountDetails editedValues={emptyObject} isEditMode={false} onInputChange={vi.fn()} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should render in edit mode', () => {
    const { container } = render(
      <AccountDetails
        editedValues={{
          2023: {
            1: '1000'
          }
        }}
        isEditMode={true}
        onInputChange={vi.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should remove year when delete button is clicked', () => {
    const { container, getAllByText } = render(
      <AccountDetails editedValues={emptyObject} isEditMode={false} onInputChange={vi.fn()} />
    );
    const deleteBtn = getAllByText('deleteYear')[0];
    fireEvent.click(deleteBtn);
    expect(mocks.mockFrom).toHaveBeenCalledWith('account_details');
    expect(container).toMatchSnapshot();
  });
});
