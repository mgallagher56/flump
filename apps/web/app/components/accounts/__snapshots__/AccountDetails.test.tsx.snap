// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AccountDetails /> > should remove year when delete button is clicked 1`] = `
<div>
  <div
    class="chakra-stack css-nn57su"
  >
    <div
      class="chakra-stack css-1igwmid"
    >
      <h3
        class="chakra-heading css-1e3tj"
      >
        2021
      </h3>
      <button
        class="chakra-button css-1ltlu47"
        type="button"
      >
        deleteYear
      </button>
    </div>
    <div
      class="d_flex flex-d_row gap_5"
    >
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          January
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          February
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          March
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          April
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          May
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          June
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          July
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          August
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          September
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          October
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          November
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          December
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
    </div>
    <div
      class="chakra-stack css-1igwmid"
    >
      <h3
        class="chakra-heading css-1e3tj"
      >
        2023
      </h3>
      <button
        class="chakra-button css-1ltlu47"
        type="button"
      >
        deleteYear
      </button>
    </div>
    <div
      class="d_flex flex-d_row gap_5"
    >
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          January
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          February
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          March
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          April
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          May
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          June
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          July
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          August
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          September
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          October
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          November
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          December
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AccountDetails /> > should render 1`] = `
<div>
  <div
    class="chakra-stack css-nn57su"
  >
    <div
      class="chakra-stack css-1igwmid"
    >
      <h3
        class="chakra-heading css-1e3tj"
      >
        2021
      </h3>
      <button
        class="chakra-button css-1ltlu47"
        type="button"
      >
        deleteYear
      </button>
    </div>
    <div
      class="d_flex flex-d_row gap_5"
    >
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          January
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          February
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          March
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          April
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          May
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          June
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          July
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          August
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          September
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          October
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          November
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          December
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
    </div>
    <div
      class="chakra-stack css-1igwmid"
    >
      <h3
        class="chakra-heading css-1e3tj"
      >
        2023
      </h3>
      <button
        class="chakra-button css-1ltlu47"
        type="button"
      >
        deleteYear
      </button>
    </div>
    <div
      class="d_flex flex-d_row gap_5"
    >
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          January
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          February
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          March
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          April
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          May
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          June
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          July
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          August
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          September
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          October
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          November
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          December
        </p>
        <p
          class="chakra-text css-au19fy"
        >
          £1,000
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AccountDetails /> > should render in edit mode 1`] = `
<div>
  <div
    class="chakra-stack css-nn57su"
  >
    <div
      class="chakra-stack css-1igwmid"
    >
      <h3
        class="chakra-heading css-1e3tj"
      >
        2021
      </h3>
      <button
        class="chakra-button css-1ltlu47"
        type="button"
      >
        deleteYear
      </button>
    </div>
    <div
      class="d_flex flex-d_row gap_5"
    >
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          January
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-January"
            hidden=""
          >
            2021-January
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="1"
                data-year="2021"
                id="2021-January"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          February
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-February"
            hidden=""
          >
            2021-February
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="2"
                data-year="2021"
                id="2021-February"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          March
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-March"
            hidden=""
          >
            2021-March
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="3"
                data-year="2021"
                id="2021-March"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          April
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-April"
            hidden=""
          >
            2021-April
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="4"
                data-year="2021"
                id="2021-April"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          May
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-May"
            hidden=""
          >
            2021-May
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="5"
                data-year="2021"
                id="2021-May"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          June
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-June"
            hidden=""
          >
            2021-June
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="6"
                data-year="2021"
                id="2021-June"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          July
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-July"
            hidden=""
          >
            2021-July
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="7"
                data-year="2021"
                id="2021-July"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          August
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-August"
            hidden=""
          >
            2021-August
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="8"
                data-year="2021"
                id="2021-August"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          September
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-September"
            hidden=""
          >
            2021-September
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="9"
                data-year="2021"
                id="2021-September"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          October
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-October"
            hidden=""
          >
            2021-October
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="10"
                data-year="2021"
                id="2021-October"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          November
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-November"
            hidden=""
          >
            2021-November
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="11"
                data-year="2021"
                id="2021-November"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          December
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2021-December"
            hidden=""
          >
            2021-December
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="12"
                data-year="2021"
                id="2021-December"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-1igwmid"
    >
      <h3
        class="chakra-heading css-1e3tj"
      >
        2023
      </h3>
      <button
        class="chakra-button css-1ltlu47"
        type="button"
      >
        deleteYear
      </button>
    </div>
    <div
      class="d_flex flex-d_row gap_5"
    >
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          January
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-January"
            hidden=""
          >
            2023-January
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="1"
                data-year="2023"
                id="2023-January"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          February
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-February"
            hidden=""
          >
            2023-February
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="2"
                data-year="2023"
                id="2023-February"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          March
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-March"
            hidden=""
          >
            2023-March
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="3"
                data-year="2023"
                id="2023-March"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          April
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-April"
            hidden=""
          >
            2023-April
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="4"
                data-year="2023"
                id="2023-April"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          May
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-May"
            hidden=""
          >
            2023-May
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="5"
                data-year="2023"
                id="2023-May"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          June
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-June"
            hidden=""
          >
            2023-June
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="6"
                data-year="2023"
                id="2023-June"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          July
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-July"
            hidden=""
          >
            2023-July
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="7"
                data-year="2023"
                id="2023-July"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          August
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-August"
            hidden=""
          >
            2023-August
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="8"
                data-year="2023"
                id="2023-August"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          September
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-September"
            hidden=""
          >
            2023-September
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="9"
                data-year="2023"
                id="2023-September"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          October
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-October"
            hidden=""
          >
            2023-October
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="10"
                data-year="2023"
                id="2023-October"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          November
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-November"
            hidden=""
          >
            2023-November
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="11"
                data-year="2023"
                id="2023-November"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
      <div>
        <p
          class="chakra-text css-au19fy"
        >
          December
        </p>
        <div
          class="d_flex flex-d_column gap_2"
        >
          <label
            for="2023-December"
            hidden=""
          >
            2023-December
          </label>
          <div>
            <div
              class="css-3a3hw0"
            >
              <input
                class="chakra-input css-v7v99c"
                data-month="12"
                data-year="2023"
                id="2023-December"
                name="amount"
                placeholder="0.00"
                type="number"
                value="1000"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
